src/djlint/formatter/utils/tag.py

```diff
@@ -157,47 +280,123 @@ def __get_tag_style(self, style: str) -> Dict:
+        block alike flow and not a script element, it is not considered as sensitive.
+
+        Returns:
+            bool: A boolean whether the element is space sensitive or not.
+        """
         if self._is_html:
             display = self.__css_display.get(self.name, self.__css_default_display)
-            # print(display)
+
             return (
                 self.is_script
                 or not display.startswith("table")
                 and display not in ["block", "list-item", "inline-block"]
             )
-        return False
 
-    @property
-    def is_hidden(self) -> bool:
-        if self._is_html:
-            return (
-                self.__css_display.get(self.name, self.__css_default_display) == "none"
-            )
         return False
 
     @property
     def is_indentation_sensitive(self) -> bool:
+        """
+        Check if tag is indentation sensitive.
+
+        Indentation sensitivity is based on element CSS flow. If element is HTML and
+        have a 'pre' alike flow it is considered as sensitive.
+
+        TODO: It seems '__tag_is_pre' have already been used in 'self.is_pre' from
+        '__init__' so it should be safe to rely on this instance attribute instead of
+        running '__tag_is_pre' again each time 'is_indentation_sensitive' is called.
+
+        Returns:
+            bool: A boolean whether the element is indentation sensitive or not.
+        """
         if self._is_html:
             return self.__tag_is_pre(self.name)
+
         return False
 
-    def __tag_is_pre(self, tag: str) -> bool:
+    @property
+    def is_hidden(self) -> bool:
+        """
+        Describe if Tag is an hidden element.
+
+        Element hidden status is determined from element CSS flow (block,
+        inline, table, etc..) as defined in HTML specifications, element CSS style rules
+        is not involved here.
+
+        Whatever hidden status is, if ``Tag._is_html`` is ``False`` this will always
+        return ``False``.
+
+        TODO: This property is a little disturbing with the ``Tag.hidden`` attribute
+        that does not seem to fullfill the same purpose.
+
+        Returns:
+            bool: True if element have a hidden status like body, meta, etc.. else
+            ``False``.
+        """
         if self._is_html:
-            return self.__css_whitespace.get(
-                tag, self.__css_default_whitespace
-            ).startswith("pre")
+            return (
+                self.__css_display.get(self.name, self.__css_default_display) == "none"
+            )
+
         return False
 
     @property
     def is_script(self) -> bool:
+        """
+        Describe if Tag is assumed as a 'script' element.
+
+        Element script status is determined from the tag name.
+
+        Whatever script status is, if ``Tag._is_html`` is ``False`` this will always
+        return ``False``.
+
+        TODO: <template> element should probably be assumed as a script element.
```

Comment:
I would prob keep template separate. The goal of this prop is to allow the contents to be formatted w/ a js formatter, and prevent djlint from touching the contents.

I think the contents of a template tag we are generally safe to format.